simbad/rotsearch/amore_search.py

```diff
@@ -378,169 +102,127 @@ def run_pdb(self, models_dir, mr_model_dir, nproc=2, shres=3.0, pklim=0.5, npic=
-                with open(dat_model, 'rb') as f_in, open(input_model, 'w') as f_out:
-                    f_out.write(zlib.decompress(base64.b64decode(f_in.read())))
-
-                try:
-                    solvent_content = matthews_coef.solvent_content(input_model,
-                                                                    cell_parameters,
-                                                                    space_group)
-                except ValueError:
-                    logger.critical("Error calculating solvent content for %s",
-                                    name)
-                    continue
+        tmp_dir = os.path.join(self.work_dir, "tmp")
+        if not os.path.isdir(tmp_dir):
+            os.makedirs(tmp_dir)
 
+        hklpck0 = self._generate_hklpck0()
+	template_hklpck1 = os.path.join(tmp_dir, "{}.hkl")
+	template_clmn0 = os.path.join(tmp_dir, "{}_spmipch.clmn")
+	template_clmn1 = os.path.join(tmp_dir, "{}.clmn")
+	template_mapout = os.path.join(tmp_dir, "{}_amore_cross.map")
+	template_table1 = os.path.join(tmp_dir, "{}_sfs.tab")
+	template_model = os.path.join(tmp_dir, "{}.pdb")
+        amore_temp_files = os.path.join(tmp_dir,
+                                        os.path.basename(self.amore_exe) + "_*${PID1}")
+
+        sol_calc = simbad.util.matthews_coef.SolventContent(cell,
+                                                            space_group)
+
+        iteration_range = range(0, len(simbad_dat_files), chunk_size)
+        for cycle, i in enumerate(iteration_range):
+            logger.info("Working on chunk %d out of %d", cycle + 1,
+                        total_chunk_cycles)
+
+            amore_files = []
+            for dat_model in simbad_dat_files[i:i + chunk_size]:
+                name = os.path.basename(dat_model.replace(".dat", ""))
+                pdb_struct = simbad.util.pdb_util.PdbStructure(dat_model)
+                solvent_content = sol_calc.calculate_from_struct(pdb_struct)
                 if solvent_content < min_solvent_content:
                     msg = "Skipping %s: solvent content is predicted to be less than %.2f"
                     logger.debug(msg, name, min_solvent_content)
                     continue
+                x, y, z, intrad = pdb_struct.integration_box
+
+                logger.debug("Generating script to perform AMORE rotation "
+                             + "function on %s", name)
+
+                pdb_model = template_model.format(name)
+                table1 = template_table1.format(name)
+                hklpck1 = template_hklpck1.format(name)
+                clmn0 = template_clmn0.format(name)
+                clmn1 = template_clmn1.format(name)
+                mapout = template_mapout.format(name)
+
+                conv_cmd = ["ccp4-python", "-c",
+                            "\"import simbad.db; simbad.db.convert_dat_to_pdb('{}', '{}')\""]
+                conv_cmd[-1] = conv_cmd[-1].format(dat_model, pdb_model)
```

Comment:
This comment is just semantics but...

Would it be easier to read if this was reversed to:

conv_cmd =  "\"import simbad.db; simbad.db.convert_dat_to_pdb('{}', '{}')\"".format(dat_model, pdb_model)
conv_cmd = ["ccp4-python", "-c", conv_cmd]

or is there a benefit to creating the list first?

Reply:
Thanks ... that was a late "hack" that's clearly gone wrong 🦆   Addressed in 4caace2